src/scripts/main.js

```diff
@@ -1,3 +1,76 @@
 'use strict';
 
-// write code here
+const refs = {
+  buttons: {
+    appendRow: document.querySelector('.append-row'),
+    removeRow: document.querySelector('.remove-row'),
+    appendCln: document.querySelector('.append-column'),
+    removeCln: document.querySelector('.remove-column'),
+  },
+  table: document.querySelector('.field'),
+  row: document.querySelector('.field tr'),
+};
+
+const MAX_ROWS = 10;
+const MIN_ROWS = 2;
+const MAX_COLUMNS = 10;
+const MIN_COLUMNS = 2;
+
+function addElement(type) {
+  if (type === 'row' && refs.table.rows.length < MAX_ROWS) {
+    const newRow = refs.table.insertRow();
+    const columnCount = refs.table.rows[0].cells.length;
+
+    for (let i = 0; i < columnCount; i++) {
+      const cell = newRow.insertCell();
+
+      cell.textContent = '';
+    }
+
+    updateButtons();
+  } else if (
+    type === 'column' &&
+    refs.table.rows[0].cells.length < MAX_COLUMNS
```

Comment:
Ensure that `refs.table.rows[0]` exists before accessing `cells.length`. If the table is empty, this could throw an error. Consider adding a check to handle cases where the table might not have any rows.